tests/testthat/_snaps/verb-select.md

```diff
@@ -181,6 +181,22 @@
       Error in `select()`:
       ! This tidyselect interface doesn't support predicates.
 
+# computed columns are not inlined away
+
+    Code
+      lf %>% mutate(z = 1) %>% arrange(x, z) %>% select(x)
+    Condition
+      Warning:
+      ORDER BY is ignored in subqueries without LIMIT
+      i Do you need to move arrange() later in the pipeline or use window_order() instead?
+    Output
+      <SQL>
+      SELECT `x`
+      FROM (
+        SELECT `df`.*, 1.0 AS `z`
+        FROM `df`
+      ) AS `q01`
```

Comment:
To me the snapshot doesn't look like it fits to the title of the test. Can you come up with a test that tests the intended behaviour more directly?

Reply:
This is just supposed to show that the subquery hasn't been removed. Do you think it should not be a snapshot test, or that should have a more descriptive title. If the latter, how about "Computed columns used to arrange are not removed by subsequent selects"?

Reply:
I think if I would come back to this test in a year and see the snapshot I wouldn't understand if the snapshot is correct or not. So, probably it shouldn't be a snapshot.

You could test the optimisation more directly, e.g.

```r
  out <- lf %>% mutate(z = 1) %>% arrange(x, z) %>% select(x)
  
  # 
  inner_query <- out$lazy_query$x
  expect_s3_class(inner_query, "lazy_select_query")
  expect_equal(
    inner_query$order_by,
    list(quo(x), quo(z)),
    ignore_formula_env = TRUE
  )

  expect_equal(op_vars(inner_query), c("x", "y", "z"))
  expect_equal(op_vars(out$lazy_query), "x")
```

and maybe also compare this to `lf %>% mutate(z = 1) %>% arrange(x, z) %>% select(x, z)`, i.e. a case where the inline optimisation was applied.

Reply:
appreciate the pointers!